flake.nix

```diff
@@ -90,104 +84,76 @@
             };
           })
       ];
-
-      projectFor = system:
+    in
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        ./pre-commit.nix
+        ./hercules-ci.nix
+      ];
+      flake = {
+        haskellModules = perSystem haskellModules;
+      };
+      systems = defaultSystems;
+      perSystem = { system, config, pkgs, ... }:
         let
-          pkgs = nixpkgsFor system;
-          pkgs' = nixpkgsFor' system;
+          project =
+            pkgs.haskell-nix.cabalProject {
+              name = "plutip-core";
+              src = ./.;
+              inputMap = {
+                "https://input-output-hk.github.io/cardano-haskell-packages" = CHaP;
+              };
+              compiler-nix-name = "ghc8107";
+
+              shell = {
+                withHoogle = true;
+                exactDeps = true;
+
+                tools.haskell-language-server = "1.5.0.0"; # Newer versions failed to build
+
+                nativeBuildInputs = with pkgs; [
+                  # Haskell Tools
+                  haskellPackages.cabal-install
```

Comment:
I 'think' you can get most of these using haskell.nix `tools`, I guess cabal-install is just cabal, the rest I'm not sure.